### ISSUE TITLE
Issue when creating CIA

### ISSUE THREAD

**Issue by phoen1x74** [phijor/ctroller#3]
I have bannertool in the /3DS/ folder, yet when I attempt "make" I get
``` make: bannertool: Command not found
Makefile:195: recipe for target '/home/ben/Documents/3ds/ctroller-master/linux/ctroller/3DS/ctroller.cia' failed
make: *** [/home/ben/Documents/3ds/ctroller-master/linux/ctroller/3DS/ctroller.cia] Error 127

```

**Comment by phoen1x74** [comment]
Nevermind. I fixed it by adding ./ to the makefile.

**Comment by phijor** [comment]
For future reference:

Unixes don't run executables in the current directories unless you explicitly tell them to. See [here](http://unix.stackexchange.com/a/4459) why that is.

You either tell them run the `bannertool` that you provided by saying `./bannertool` (what you did), or you place `bannertool` in a directory that is in your [`$PATH`](https://en.wikipedia.org/wiki/PATH_(variable)#Unix_and_Unix-like) (what I'd recommend).

**Comment by yellomonkey** [comment]
-e \e[1mBuilding linux executable...\e[0m
make[1]: Entering directory `/home/maya/Desktop/ctroller-master/linux'
Creating directories
Beginning release build
Compiling: src/main.c -> build/release/main.o
src/main.c: In function ‘main’:
src/main.c:134:12: warning: missing initializer for field ‘version’ of ‘struct hidinfo’ [-Wmissing-field-initializers]
     struct hidinfo hid = {};
            ^
In file included from include/ctroller.h:6:0,
                 from src/main.c:31:
include/hid.h:69:14: note: ‘version’ declared here
     uint16_t version;
              ^
In file included from src/main.c:31:0:
include/ctroller.h:18:37: error: ‘VERSION_MAJOR’ undeclared (first use in this function)
 #define CTROLLER_VERSION MAKEBCDVER(VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH)
                                     ^
include/ctroller.h:12:7: note: in definition of macro ‘MAKEBCDVER’
     ((major & 0xf) << 8 | (minor & 0xf) << 4 | (patch & 0xf))
       ^
src/main.c:152:36: note: in expansion of macro ‘CTROLLER_VERSION’
                 if (hid.version != CTROLLER_VERSION) {
                                    ^
include/ctroller.h:18:37: note: each undeclared identifier is reported only once for each function it appears in
 #define CTROLLER_VERSION MAKEBCDVER(VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH)
                                     ^
include/ctroller.h:12:7: note: in definition of macro ‘MAKEBCDVER’
     ((major & 0xf) << 8 | (minor & 0xf) << 4 | (patch & 0xf))
       ^
src/main.c:152:36: note: in expansion of macro ‘CTROLLER_VERSION’
                 if (hid.version != CTROLLER_VERSION) {
                                    ^
include/ctroller.h:18:52: error: ‘VERSION_MINOR’ undeclared (first use in this function)
 #define CTROLLER_VERSION MAKEBCDVER(VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH)
                                                    ^
include/ctroller.h:12:28: note: in definition of macro ‘MAKEBCDVER’
     ((major & 0xf) << 8 | (minor & 0xf) << 4 | (patch & 0xf))
                            ^
src/main.c:152:36: note: in expansion of macro ‘CTROLLER_VERSION’
                 if (hid.version != CTROLLER_VERSION) {
                                    ^
include/ctroller.h:18:67: error: ‘VERSION_PATCH’ undeclared (first use in this function)
 #define CTROLLER_VERSION MAKEBCDVER(VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH)
                                                                   ^
include/ctroller.h:12:49: note: in definition of macro ‘MAKEBCDVER’
     ((major & 0xf) << 8 | (minor & 0xf) << 4 | (patch & 0xf))
                                                 ^
src/main.c:152:36: note: in expansion of macro ‘CTROLLER_VERSION’
                 if (hid.version != CTROLLER_VERSION) {
                                    ^
make[2]: *** [build/release/main.o] Error 1
make[1]: *** [release] Error 2
make[1]: Leaving directory `/home/maya/Desktop/ctroller-master/linux'
make: *** [linux] Error 2

**Comment by yellomonkey** [comment]
i did all the steps but i'm getting an error 
i hope you would kindly help me

**Comment by yellomonkey** [comment]
could ya please give the .cia file please???

**Comment by phijor** [comment]
Hi @yellomonkey, you probably did not build `ctroller` from a git-repository. See issue #2 why you get that error. To build `ctroller`, you either clone the repository and let it build from there or wait for the next commit that should fix out-of-git building/versioning.

If you want to download the latest release of `ctroller`, just got to [the release tab](https://github.com/phijor/ctroller/releases/latest), you'll find a `.cia` file there.

Thank you for reporting, but the next time I'd urge you to [open a new issue](https://github.com/phijor/ctroller/issues/new) on the tracker, that makes things easier to organize.

**Comment by yellomonkey** [comment]
thanks dude
i'll try my best to organize things
awesome app and it works like a charm
